### PR TITLE
change the pod  disruption budget from 1 to 25%

### PR DESCRIPTION
**Pull request by weatherhog**
as requested by @LolloneS the pdb changed from absolut number to percentage. The original 1 is replaced with 25 %

### CODE REVIEW

**Comment by LolloneS** [comment]
@paurosello @njuettner does 25% look fine for you as a replacement for `maxUnavailable: 1`?

**Comment by njuettner** [comment]
what happens if you have 3 replicas, this would mean it would never kill a pod or am I wrong?

**Comment by njuettner** [comment]
my bad 🤦🏻 it's unavailable. yeah make sense

**Comment by njuettner** [comment]
but in general I would like to have a consistent way how we define unavailable, what is the reason from switching to percent?

File: helm/coredns-app/values.schema.json [hunk 1]
```diff
@@ -191,7 +191,7 @@
                     "type": "object",
                     "properties": {
                         "maxUnavailable": {
-                            "type": "integer"
+                            "type": "string"
```
Comment (ubergesundheit) [inline helm/coredns-app/values.schema.json:194]:
```suggestion
                            "type": ["string", "integer"]
```

**Comment by njuettner** [comment]
@LolloneS we tuned the node-operator for taking care of killing pods and not waiting anymore than minutes. this was an outcome of the adidas cluster issues we had. The node-operator does include the core K8s draining logic and it will try to evict pods for a while if it's not possible it will terminate the instance anyway, we're hoping to improve the stability of upgrades.

But I'm totally fine to merge 👍🏻

**Comment by LolloneS** [comment]
>@LolloneS we tuned the node-operator for taking care of killing pods and not waiting anymore than minutes.

I think there is a difference though. This is related to _graceful_ shutdown and in general it's good not to have absolute numbers but percentages. 


As a side note/opinion, if we say "just kill the node if you can't drain it for 2 minutes" in `node-operator` we might be missing out on the possibility that by waiting e.g. 10 minutes we can gracefully kill the node because a PDB becomes satisfied. 

So in my opinion:
- `node-operator` should use the default K8s draining logic, but allow for some minutes so that a node can come up and get some pods scheduled so that PDBs become satisfied (e.g. 10mins so that AWS-CNI can provide the needed IPs)
- `coredns` should use percentages to have reasonable PDBs at any scale

WDYT?

**Review by LolloneS — APPROVED** [review]

**Review by ubergesundheit — APPROVED** [review]

**Comment by mcharriere** [comment]
please update the changelog

**Comment by LolloneS** [comment]
@mcharriere done, will merge on monday if nothing else comes up

**Pull request merged**